app/src/debug/java/org/simple/clinic/FakeDataGenerationReceiver.kt

```diff
@@ -76,11 +76,11 @@ class FakeDataGenerationReceiver : BroadcastReceiver() {
   private fun generateFakeData(recordsToGenerate: Int): Completable {
     return Completable.fromAction {
 
-      val user = userSession.loggedInUserImmediate()!!
-      val currentFacility = facilityRepository.currentFacilityImmediate(user)!!
-      val otherFacility = anyFacilityExceptCurrent(user, currentFacility)
+      val currentUser = user.get()
+      val currentFacility = facilityRepository.currentFacilityImmediate(currentUser)!!
```

Comment:
This could also be injected, but since it is not real production code, I am fine with leaving it as is.

Reply:
I left this as is because `facilityRepository` was anyway being used elsewhere in the class. So injecting `facility` directly wouldn't have simplified much.
I can make this change though.

Reply:
Not necessary.